pkg/build/node/node.go

```diff
@@ -446,23 +449,36 @@ func (c *BuildContext) prePullImages(dir, containerID string) error {
 		return errors.Wrap(err, "failed to make images dir")
 	}
 
-	pulled := []string{}
+	g, _ := errgroup.WithContext(context.Background())
+	pulledImages := make(chan string, len(requiredImages))
 	for i, image := range requiredImages {
-		if !builtImages.Has(image) {
-			fmt.Printf("Pulling: %s\n", image)
-			err := docker.Pull(image, 2)
-			if err != nil {
-				return err
-			}
-			// TODO(bentheelder): generate a friendlier name
-			pullName := fmt.Sprintf("%d.tar", i)
-			pullTo := path.Join(imagesDir, pullName)
-			err = docker.Save(image, pullTo)
-			if err != nil {
-				return err
+		i, image := i, image // https://golang.org/doc/faq#closures_and_goroutines
+		g.Go(func() error {
```

Comment:
what about using current concurrent functions https://github.com/kubernetes-sigs/kind/blob/master/pkg/concurrent/concurrent.go?

There are some examples in the code of using it in Ben´s commit https://github.com/kubernetes-sigs/kind/commit/110dc82f093cdcb7835a01ba85c1ce714083f434

Reply:
Thanks, didn't know about these. Changed! :slightly_smiling_face:

Reply:
this other library is really neat though! will need to revisit this sometime, thanks for the pointer :-)